Fix/design tweaks

hakjeri opened the pull request.

tasola reviewed (changes requested):
I will review the code soon, but first I'm going through the rendered version. Some questions:

1. Error on **Filter page**  `Using kebab-case for css properties in objects is not supported. Did you mean marginLeft?`
2. It seems like we render the "Level" icon (the three bars) in the rare cases where we don't have a level. I don't think you can set a relation to mandatory in Strapi (...), therefore I think we can't count on it being defined in all cases either...
3. (Not a big deal, so I don' think it's worth looking into): On pages with small content, the footer seems to follow the content's height, pushing it upwards on the page, creating a gap of white beneath the footer.
4. (Again, minor thing): Did you talk about hover states on buttons? Right now we only have the cursor changing, but some color indication of similar would be nice as well imo.
5. I'm having a hard time finding a solid way to reproduce it, but I get this error sometimes when fooling around on the **filter page**: `Unsupported style property @media (min-width: 768px). Did you mean @media (minWidth: 768px)?`
6. (I noticed that we get 500 when sorting on Level, but that's obviously not introduced in this PR. I created a card for it: https://trello.com/c/9eDTgcxb/168-error-when-sorting-on-level)
7. Sometimes when I enter the **start page** (again, not every time (...)) I get: `The pseudo class ":nth-child" is potentially unsafe when doing server-side rendering. Try changing it to ":nth-of-type". `

I think it looks great though! Really nice improvements which really make a big difference. The mobile view is also beautiful. Good job! 🌟

tasola commented on src/components/ButtonLink/styles.ts:
```diff
@@ -11,7 +11,7 @@ export const ButtonLink = styled(Link, {
 })<LinkProps & ButtonLinkProps>`
   display: inline-block;
-  padding: 7px 24px;
+  padding: 11px 24px;
```
Any reason for these values not to be in `rem` btw?

hakjeri replied:
No, good catch, it's fixed!

tasola commented on src/components/TabGroup/TabGroup.tsx:
```diff
@@ -195,6 +209,20 @@ const TabGroup = ({ selectedKeywords, selectedAuthors }: Props) => {
+                ...blocks,
+                ...lecture.attributes.Blocks.data,
+              ],
+              [] as Data<Pick<BlockOneLevelDeep, 'DurationInMinutes'>>[]
```
Nice

tasola reviewed (commented):
The code looks great!

hakjeri commented:
> I will review the code soon, but first I'm going through the rendered version. Some questions:
> 
>     1. Error on **Filter page**  `Using kebab-case for css properties in objects is not supported. Did you mean marginLeft?`
> 
>     2. It seems like we render the "Level" icon (the three bars) in the rare cases where we don't have a level. I don't think you can set a relation to mandatory in Strapi (...), therefore I think we can't count on it being defined in all cases either...
> 
>     3. (Not a big deal, so I don' think it's worth looking into): On pages with small content, the footer seems to follow the content's height, pushing it upwards on the page, creating a gap of white beneath the footer.
> 
>     4. (Again, minor thing): Did you talk about hover states on buttons? Right now we only have the cursor changing, but some color indication of similar would be nice as well imo.
> 
>     5. I'm having a hard time finding a solid way to reproduce it, but I get this error sometimes when fooling around on the **filter page**: `Unsupported style property @media (min-width: 768px). Did you mean @media (minWidth: 768px)?`
> 
>     6. (I noticed that we get 500 when sorting on Level, but that's obviously not introduced in this PR. I created a card for it: https://trello.com/c/9eDTgcxb/168-error-when-sorting-on-level)
> 
>     7. Sometimes when I enter the **start page** (again, not every time (...)) I get: `The pseudo class ":nth-child" is potentially unsafe when doing server-side rendering. Try changing it to ":nth-of-type". `
> 
> 
> I think it looks great though! Really nice improvements which really make a big difference. The mobile view is also beautiful. Good job! 🌟

Excellent feedback, and thank you :)

1. Fixed!
2. Fixed!
3. Yes, this requires a bit of thought. I think we can add some `min-height` to either main or body, or perhaps make the footer stick to the bottom. But it'll require some thought so I think it's best left for now.
4. No, we forgot the hover states, good catch! Fixed!
5. This is due to some issue with using MUI and styled-components mixed. It seems to work, and there are no warnings on full-reloads, but navigation to the filter page seems to trigger this. I tried adding MUI-styled breakpoints, but while the warnings disappeared, the styling broke instead. Needs further investigation, I added a ticket: https://trello.com/c/XX3T87DF/169-mui-and-styled-components-clash-on-filter-page
6. Great!
7. Yes, it works just as well with `nth-of-type` in this case, so I changed it.

tasola reviewed (approved):
Great job! 🌟
